### PR TITLE
Threaded _fmpz_vec_get/set_fft; simplify by not returning bit count

### PR DESCRIPTION
**Pull request by fredrik-johansson**
Doing FFT conversions in parallel speeds up big multiplications 5-15%.

Timings for ``fmpz_poly_mul_SS`` with 8 threads:

         n       bits         old       new
       100         40    4.42e-05  4.36e-05   1.014x speedup
       100        160    5.92e-05  5.88e-05   1.007x speedup
       100        640    0.000134  0.000133   1.008x speedup
       100       2560    0.000708  0.000707   1.001x speedup
       100      10240     0.00438   0.00436   1.005x speedup
       100      40960      0.0224    0.0224   1.000x speedup
       100     163840        0.11     0.109   1.009x speedup
       200         40    0.000118  0.000119   0.992x speedup
       200        160    0.000155  0.000148   1.047x speedup
       200        640    0.000259  0.000258   1.004x speedup
       200       2560     0.00108   0.00108   1.000x speedup
       200      10240     0.00652    0.0065   1.003x speedup
       200      40960      0.0325    0.0319   1.019x speedup
       200     163840       0.168     0.166   1.012x speedup
       400         40    0.000291  0.000284   1.025x speedup
       400        160    0.000324  0.000317   1.022x speedup
       400        640    0.000549  0.000546   1.005x speedup
       400       2560     0.00231   0.00226   1.022x speedup
       400      10240      0.0133    0.0132   1.008x speedup
       400      40960       0.069     0.068   1.015x speedup
       400     163840        0.34     0.334   1.018x speedup
       800         40     0.00079   0.00072   1.097x speedup
       800        160     0.00076    0.0007   1.086x speedup
       800        640     0.00126   0.00125   1.008x speedup
       800       2560     0.00435   0.00434   1.002x speedup
       800      10240      0.0239    0.0238   1.004x speedup
       800      40960       0.125      0.12   1.042x speedup
       800     163840       0.564     0.552   1.022x speedup
      1600         40     0.00188   0.00179   1.050x speedup
      1600        160     0.00187   0.00177   1.056x speedup
      1600        640     0.00255    0.0025   1.020x speedup
      1600       2560       0.008     0.008   1.000x speedup
      1600      10240      0.0473     0.047   1.006x speedup
      1600      40960       0.235     0.227   1.035x speedup
      1600     163840       1.082     1.046   1.034x speedup
      3200         40      0.0052    0.0049   1.061x speedup
      3200        160      0.0053    0.0049   1.082x speedup
      3200        640      0.0052    0.0048   1.083x speedup
      3200       2560      0.0139    0.0129   1.078x speedup
      3200      10240       0.085     0.079   1.076x speedup
      3200      40960       0.342     0.324   1.056x speedup
      3200     163840       1.525     1.464   1.042x speedup
      6400         40      0.0176    0.0154   1.143x speedup
      6400        160      0.0178    0.0157   1.134x speedup
      6400        640      0.0176    0.0156   1.128x speedup
      6400       2560       0.037     0.034   1.088x speedup
      6400      10240       0.164     0.153   1.072x speedup
      6400      40960       0.657     0.623   1.055x speedup
      6400     163840        3.05     2.898   1.052x speedup
     12800         40       0.083     0.072   1.153x speedup
     12800        160       0.083     0.073   1.137x speedup
     12800        640       0.082     0.072   1.139x speedup
     12800       2560       0.082     0.073   1.123x speedup
     12800      10240       0.292     0.271   1.077x speedup
     12800      40960       1.158     1.082   1.070x speedup
     12800     163840       5.483      5.12   1.071x speedup
     25600         40       0.328     0.301   1.090x speedup
     25600        160       0.328     0.301   1.090x speedup
     25600        640       0.327     0.302   1.083x speedup
     25600       2560       0.329       0.3   1.097x speedup
     25600      10240       0.735     0.688   1.068x speedup
     25600      40960       2.514     2.339   1.075x speedup
     25600     163840      11.644    10.887   1.070x speedup
     51200         40       1.499     1.402   1.069x speedup
     51200        160       1.511     1.401   1.079x speedup
     51200        640        1.51     1.404   1.075x speedup
     51200       2560       1.525     1.417   1.076x speedup
     51200      10240       1.524     1.421   1.072x speedup
     51200      40960       5.152     4.811   1.071x speedup
     51200     163840      24.024    22.523   1.067x speedup

### CODE REVIEW

File: fmpz_poly/mullow_SS.c [hunk 1]
```diff
@@ -37,8 +37,15 @@ void _fmpz_poly_mullow_SS(fmpz * output, const fmpz * input1, slong len1,
     loglen2 = FLINT_CLOG2(len2);
     n = (WORD(1) << (loglen - 2));
 
-    size1 = _fmpz_vec_max_limbs(input1, len1); 
-    size2 = _fmpz_vec_max_limbs(input2, len2);
+    bits1 = _fmpz_vec_max_bits(input1, len1);
```
Comment (wbhart) [inline fmpz_poly/mullow_SS.c:40]: I don't understand why it is better to use the presumably more expensive vec_max_bits function given that we convert it to a number of limbs below. The actual bits are only used to determine if ant coefficients were signed right? But we get that info from get_fft below, so where is the saving?

Comment (wbhart) [reply fmpz_poly/mullow_SS.c:40]: Nevermind, I see that the get_fft function now no longer needs to return the sign. Yeah this is fine.

**Comment by wbhart** [comment]
Yeah this looks good. I'll merge when appveyor says it is happy.

**Comment by fredrik-johansson** [comment]
Note that ``mul`` always computes the number of bits, so if the bit counting were a bottleneck, we'd be better off just having a version of ``mul_SS`` where we can pass in the precomputed number of bits (in which case counting bits in ``get_fft`` would be redundant). I didn't make this change because ``mul_SS`` mostly cares about huge coefficients, where bit counting isn't going to be a bottleneck; taking bit counting out of ``get_fft`` was mostly intended as a code simplification.

It could be worth having a version of ``mul_KS`` where we pass in the bits, on the other hand.

**Pull request merged**